README.md

```diff
@@ -39,25 +39,108 @@ in `psplash/config` to point to this file and rebuild the gadget.
 
 To turn off the splash screen completely please edit `configs/core/cmdline.txt`
 and remove the `splash` and the `vt.handoff=2` keywords from the default kernel
-commandline.
-
+command line.
 
-## Building
+## Branches
 
-This gadget snap can optionally be cross built on an amd64 machine. To do so,
-just run `snapcraft` with an appropriate `--target-arch` switch, and
-`--destructive-mode` in the top level of the source tree after cloning it and
-selecting the appropriate branch:
-
-    $ sudo snap install snapcraft --classic
-    $ git clone https://github.com/snapcore/pi-gadget
-    $ cd pi-gadget
-    $ git checkout 18-arm64
-    $ sudo snapcraft clean --destructive-mode
-    $ sudo snapcraft snap --target-arch=arm64 --destructive-mode
-
-The branches included are:
+This repository contains the following branches for Ubuntu Core versions and
+the two Raspberry Pi architectures:
 
+* 20-arm64 - the branch for Core 20 on arm64 (**default**)
+* 20-armhf - the branch for Core 20 on armhf
 * 18-arm64 - the branch for Core 18 on arm64
 * 18-armhf - the branch for Core 18 on armhf
-* classic - the branch for Ubuntu (universal gadget)
+
+## Building
+
+There two options for building the gadget snap: cross building and native
+building.
+
+### Cross building
+
+This is likely the most convenient and performant build method as the gadget is
+built within a container on the host machine.
+
+#### Prerequisites
+
+- An Ubuntu host (20.04 or newer is recommended)
+- [Snapcraft](https://snapcraft.io/docs/snapcraft-overview)
+
+To build the gadget snap, switch to the appropriate branch and simply
+run the `snapcraft` command:
+
+```bash
+$ git clone https://github.com/snapcore/pi-gadget
+$ cd pi-gadget
+$ git checkout 20-armhf
+$ snapcraft
+[...]
+Snapped pi_20-1_armhf.snap
+```
+
+By default, _snapcraft_ attempts to build the gadget snap in a
+[Multipass](https://multipass.run/) container, isolating the host system from
+the build system. [Building on LXD](https://snapcraft.io/docs/build-on-lxd) is
+another option that can be faster, especially when iterating over builds.
+
+If Multipass or LXD is not already installed, _Snapcraft_ will install the
+appropriate packages and run through their setup before building the gadget.
+
+Both Multipass and LXD allow for the build architecture to differ from the
+_run-on_ architecture, as defined by the `architecture` stanza in the
+_snapcraft.yaml_ for the gadget snap:
+
+```yaml
+architecture
+  - build-on: [amd64, armhf]
+    run-on: armhf
+```
+
+See [Architectures](https://snapcraft.io/docs/architectures) for more details
+on defining architectures and [Image
+building](https://ubuntu.com/core/docs/board-enablement#heading--image-building) 
+for instructions on how to build a bootable image that includes the gadget snap.
+
+### Native building
```

Comment:
I don't mean to nitpick, but I think really we could make this more clear by separating how you build not by native vs cross, but rather by _snapcraft-managed_ or _manual-managed_ build environments. 

Here's where the various builds fall into these two categories for example:
- build the gadget snap on a classic ubuntu amd64 desktop -> snapcraft managed 
- build the gadget snap on a classic ubuntu arm64 pi desktop -> snapcraft managed 
- build the gadget snap on an ubuntu core pi machine -> manual managed via lxd (or multipass maybe, I dunno if KVM support on pi is working yet, it's not the past few times I checked)
- build the gadget snap on some fast specific build machine -> manual managed via lxd or whatever

Then the tutorial essentially has your `Cross building` section become the snapcraft-managed build environment where you install the classic snapcraft snap + either lxd or multipass and just do `snapcraft` and get a working snap, or you describe how folks can setup a manually managed instance via LXD like one would do on UC20 (or wherever they want to run LXD), pointing out the most important bits that 1) the managed instance has the same distro series as declared in the gadget's snapcraft.yaml `base` setting, which will change depending on if you are building the 20 gadget or the 18 gadget for example and 2) that snapcraft is called with `--destructive-mode` flag.

Reply:
Hi, did anyone consider this sort of refactoring to this doc?